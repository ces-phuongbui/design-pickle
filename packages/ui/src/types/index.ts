interface Image {
  id: string;
  documentId: string;
  url: string;
  alternativeText: string | null;
  name: string;
}

export enum ButtonType {
  DEFAULT = "default",
  DESTRUCTIVE = "destructive",
  OUTLINE = "outline",
  SECONDARY = "secondary",
  GHOST = "ghost",
  LINK = "link",
}

export interface IButton {
  label: string;
  href?: string;
  type: ButtonType;
  color?: string;
}

export type ICarouselItem = {
  id: string | number;
  image: Image;
  name: string;
};

export type CardFeature = {
  id: number | string;
  title: string;
  cta: IButton;
  backgroundImage?: Image;
  description?: string;
};

export type CardStat = {
  value: string | number;
  label?: string;
  subLabel: string;
};

export type ComponentType =
  | "section.introduce-header"
  | "section.social-carousel"
  | "section.features"
  | "section.start";

interface Base<T extends ComponentType, D extends {} = {}> {
  __component: T;
  id: string;
  createdAt: string;
  updatedAt: string;
  data?: D;
}

export interface NavLink {
  href: string;
  text: string;
  isExternal: boolean;
  isPrimary: boolean;
}

export type Block = Introduce | SocialCarousel | Features | Stat;

export interface Introduce extends Base<"section.introduce-header"> {
  title: string;
  description: string;
  button: IButton;
}

export interface SocialCarousel extends Base<"section.social-carousel"> {
  bannerImage: Image;
  logos: ICarouselItem[];
}

export interface Features extends Base<"section.features"> {
  title: string;
  subTitle?: string;
  cards: CardFeature[];
}

export interface Stat extends Base<"section.start"> {
  title: string;
  subTitle: string;
  cards: CardStat[];
}
