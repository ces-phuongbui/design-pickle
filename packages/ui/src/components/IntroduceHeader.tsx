import { Introduce } from "../types";
import { Button } from "./ui/button";
import Link from "next/link";

const IntroduceHeader = ({ title, description, button }: Introduce) => {
  return (
    <section className="relative w-full bg-black text-white m-auto">
      <div className="container mx-auto px-4 py-16 md:py-24">
        <div className="mx-auto flex ">
          <h1 className="mb-6 text-4xl max-w-xl font-bold leading-tight tracking-tight md:text-5xl lg:text-6xl">
            {title}
          </h1>
          <div className="w-2/4">
            <p className="mb-8 text-lg text-gray-300 md:text-xl max-w-[80%]">
              {description}
            </p>
            <Link href={button.href ?? "/"}>
              <Button
                variant={button.type || "default"}
                className={`text-base font-medium ${
                  button.color ? "bg-[`${button.color}`] text-black" : ""
                }`}
                style={
                  button.color ? { backgroundColor: button.color } : undefined
                }
              >
                {button.label}
              </Button>
            </Link>
          </div>
        </div>
      </div>
    </section>
  );
};

export default IntroduceHeader;
