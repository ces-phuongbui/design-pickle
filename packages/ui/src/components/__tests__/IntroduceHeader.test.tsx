import { ButtonType, Introduce } from "@repo/ui/types/index";
import { render, screen } from "@testing-library/react";
import IntroduceHeader from "../IntroduceHeader";

describe("IntroduceHeader", () => {
  const mockProps: Introduce = {
    __component: "section.introduce-header",
    id: "1",
    createdAt: "2023-10-01T00:00:00.000Z",
    updatedAt: "2023-10-01T00:00:00.000Z",
    title: "Welcome to Our Platform",
    description: "This is a description of our platform.",
    button: {
      label: "Get Started",
      href: "/get-started",
      type: ButtonType.DEFAULT,
      color: "#D5FF02",
    },
  };

  it("renders correctly and matches snapshot", () => {
    const { container } = render(<IntroduceHeader {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the title correctly", () => {
    render(<IntroduceHeader {...mockProps} />);
    const title = screen.getByText("Welcome to Our Platform");
    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe("H1");
  });

  it("renders the description correctly", () => {
    render(<IntroduceHeader {...mockProps} />);
    const description = screen.getByText(
      "This is a description of our platform.",
    );
    expect(description).toBeInTheDocument();
    expect(description.tagName).toBe("P");
  });

  it("renders the button link correctly", () => {
    render(<IntroduceHeader {...mockProps} />);
    const link = screen.getByRole("link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/get-started");
  });
});
