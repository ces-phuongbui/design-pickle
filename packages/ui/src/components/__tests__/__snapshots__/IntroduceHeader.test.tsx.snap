// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroduceHeader renders correctly and matches snapshot 1`] = `
<div>
  <section
    class="relative w-full bg-black text-white m-auto"
  >
    <div
      class="container mx-auto px-4 py-16 md:py-24"
    >
      <div
        class="mx-auto flex "
      >
        <h1
          class="mb-6 text-4xl max-w-xl font-bold leading-tight tracking-tight md:text-5xl lg:text-6xl"
        >
          Welcome to Our Platform
        </h1>
        <div
          class="w-2/4"
        >
          <p
            class="mb-8 text-lg text-gray-300 md:text-xl max-w-[80%]"
          >
            This is a description of our platform.
          </p>
          <a
            href="/get-started"
          >
            <button
              class="inline-flex items-center justify-center rounded-md transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background shadow hover:bg-primary/90 h-10 py-2 px-4 text-base font-medium bg-[\`\${button.color}\`] text-black"
              style="background-color: rgb(255, 0, 0);"
            >
              Get Started
            </button>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
